src/jit-dasm/jit-dasm.cs

```diff
@@ -1,723 +0,0 @@
-using System.Runtime.Loader;
-using System.Linq;
-using System.Text;
-
-namespace ManagedCodeGen
-{
-    public enum CodeGenerator
-    {
-        Crossgen,
-        Crossgen2
-    }
-
-    // Define options to be parsed
-    public class Config
-    {
-        private ArgumentSyntax _syntaxResult;
-        private string _altjit = null;
-        private string _crossgenExe = null;
-        private string _jitPath = null;
-        private string _rootPath = null;
-        private string _fileName = null;
-        private IReadOnlyList<string> _assemblyList = Array.Empty<string>();
-        private bool _wait = false;
-        private bool _recursive = false;
-        private IReadOnlyList<string> _methods = Array.Empty<string>();
-        private IReadOnlyList<string> _platformPaths = Array.Empty<string>();
-        private bool _dumpGCInfo = false;
-        private bool _dumpDebugInfo = false;
-        private bool _verbose = false;
-        private bool _noDiffable = false;
-        private CodeGenerator _codeGenerator;
-
-        public Config(string[] args)
-        {
-            _syntaxResult = ArgumentSyntax.Parse(args, syntax =>
-            {
-                syntax.DefineOption("altjit", ref _altjit, "If set, the name of the altjit to use (e.g., clrjit_win_arm64_x64.dll).");
-                syntax.DefineOption("c|crossgen", ref _crossgenExe, "The crossgen or crossgen2 compiler exe.");
-                syntax.DefineOption("j|jit", ref _jitPath, "The full path to the jit library.");
-                syntax.DefineOption("o|output", ref _rootPath, "The output path.");
-                syntax.DefineOption("f|file", ref _fileName, "Name of file to take list of assemblies from. Both a file and assembly list can be used.");
-                syntax.DefineOption("gcinfo", ref _dumpGCInfo, "Add GC info to the disasm output.");
-                syntax.DefineOption("debuginfo", ref _dumpDebugInfo, "Add Debug info to the disasm output.");
-                syntax.DefineOption("v|verbose", ref _verbose, "Enable verbose output.");
-                syntax.DefineOption("nodiffable", ref _noDiffable, "Generate non-diffable asm (pointer values will be left in output).");
-                var waitArg = syntax.DefineOption("w|wait", ref _wait, "Wait for debugger to attach.");
-                waitArg.IsHidden = true;
-
-                syntax.DefineOption("r|recursive", ref _recursive, "Scan directories recursively.");
-                syntax.DefineOptionList("p|platform", ref _platformPaths, "Path to platform assemblies");
-                var methodsArg = syntax.DefineOptionList("m|methods", ref _methods,
-                    "List of methods to disasm.");
-                methodsArg.IsHidden = true;
-
-                // Warning!! - Parameters must occur after options to preserve parsing semantics.
-
-                syntax.DefineParameterList("assembly", ref _assemblyList, "The list of assemblies or directories to scan for assemblies.");
```

Comment:
I don't think this behavior is preserved. From my understanding, the parameter list here means that "unnamed" arguments will go into this parameter, e.g. `jit-dasm --option1 value1 --option2 value2 assembly1 assembly2` will put `assembly1` and `assembly2` into this list. However, on this PR, when I try:
```
jit-diff diff --output C:\dev\temp\diffs\ --tag r2rchange --base C:\dev\dotnet\runtime\artifacts\bin\coreclr\windows.x64.Checked\ --base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\ --core_root "C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files" --crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe -v
```

(after building a checked build in C:\dev\dotnet\runtime, and copying Core_Root to Core_Root_files) I get

```scala
❯ jit-diff diff --output C:\dev\temp\diffs\ --tag r2rchange --base C:\dev\dotnet\runtime\artifacts\bin\coreclr\windows.x64.Checked\ --base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\ --core_root "C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files" --crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe -v
No assemblies specified; defaulting to corelib
After setting defaults:
--core_root C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files
--crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe
--output C:\dev\temp\diffs\
--base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\
Beginning Crossgen CodeSize Diffs for System.Private.CoreLib.dll
Scanning: C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files\System.Private.CoreLib.dll
Started dasm command "--platform C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files --verbose --crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe --jit C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\clrjit.dll --output C:\dev\temp\diffs\r2rchange\base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files\System.Private.CoreLib.dll"
Dasm command: jit-dasm --platform C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files --verbose --crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe --jit C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\clrjit.dll --output C:\dev\temp\diffs\r2rchange\base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files\System.Private.CoreLib.dll
Description:
  Managed codegen diff tool (crossgen/AOT)

Usage:
  jit-dasm [options]

Options:
  --altjit <altjit>          If set, the name of the altjit to use (e.g., clrjit_win_arm64_x64.dll)
  -c, --crossgen <crossgen>  The crossgen or crossgen2 compiler exe. []
  -j, --jit <jit>            The full path to the jit library []
  -o, --output <output>      The output path
  -f, --file <file>          Name of file to take list of assemblies from. Both a file and assembly list can be used
  --gcinfo                   Add GC info to the disasm output
  --debuginfo                Add Debug info to the disasm output
  --verbose                  Enable verbose output
  --nodiffable               Generate non-diffable asm (pointer values will be left in output)
  -r, --recursive            Search directories recursively
  -p, --platform <platform>  Path to platform assemblies
  -m, --methods <methods>    List of methods to disasm
  --assembly <assembly>      The list of assemblies or directories to scan for assemblies
  -w, --wait                 Wait for debugger to attach
  -v, --version              Show version information
  -?, -h, --help             Show help and usage information


Unrecognized command or argument 'C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files\System.Private.CoreLib.dll'.

Dasm command "jit-dasm --platform C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files --verbose --crossgen C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\crossgen2\crossgen2.exe --jit C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root\clrjit.dll --output C:\dev\temp\diffs\r2rchange\base C:\dev\dotnet\runtime\artifacts\tests\coreclr\windows.x64.Checked\Tests\Core_Root_files\System.Private.CoreLib.dll" returned with 1 failures
```

Reply:
After fixing the above there is another issue with obtaining a full set of jit disasm with crossgen2, introduced by me recently. https://github.com/dotnet/runtime/pull/80371 should fix it.

Reply:
Could you push that change to PR branch?